### PR TITLE
Feature/new ux   home screen

### PR DESCRIPTION
**Pull request by mellowgeek**

### CODE REVIEW

**Comment by mellowgeek** [comment]
Screenshot_1588896830
Screenshot_1588897319

File: app/src/main/java/org/coepi/android/ui/home/HomeViewModel.kt [hunk 1]
```diff
@@ -21,13 +22,27 @@ import org.coepi.android.ui.symptoms.SymptomsFragmentDirections.Companion.action
...
 ) : ViewModel() {
 
     val versionString: LiveData<String> =
         just(resources.getString(home_version, envInfos.appVersionString()))
+            .toLiveData()
+
+    val newAlerts: LiveData<Boolean> = alertsRepo.alerts
+        .map { newAlerts(it.size) }
+        .startWith(newAlerts(0))
+        .observeOn(AndroidSchedulers.mainThread())
```
Comment (i-schuetz) [inline app/src/main/java/org/coepi/android/ui/home/HomeViewModel.kt:36]: You can use static import here

File: app/src/main/res/layout/fragment_home.xml [hunk 2]
```diff
@@ -19,141 +20,152 @@
...
-                    app:layout_constraintStart_toStartOf="parent"
-                    android:layout_marginStart="16dp"
-                    android:layout_marginTop="18dp"
-                    android:src="@drawable/ic_contact_alert_icon" />
+                    android:layout_marginTop="@dimen/default_line_height"
+                    android:layout_marginStart="48dp"
+                    android:text="@{vm.title}"
+                    android:textSize="17sp"
+                    android:textColor="@color/red"
+                    android:visibility="@{vm.newAlerts ? View.VISIBLE : View.GONE}"
```
Comment (i-schuetz) [inline app/src/main/res/layout/fragment_home.xml:112]: You can use `isVisible` property to bind directly to a boolean (see `XmlBindingAdapters`).

File: app/src/main/res/layout/fragment_home.xml [hunk 3]
```diff
@@ -19,141 +20,152 @@
...
+                android:id="@+id/alert_icon"
+                app:layout_constraintStart_toStartOf="@id/contact_alerts_card"
+                app:layout_constraintTop_toTopOf="@id/contact_alerts_card"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:src="@drawable/ic_alert_circle"
+                android:elevation="2dp"
+                android:translationY="-8dp"
+                android:translationX="-8dp"
+                android:visibility="@{vm.newAlerts ? View.VISIBLE : View.GONE}"/>
```
Comment (i-schuetz) [inline app/src/main/res/layout/fragment_home.xml:146]: You can use isVisible property to bind directly to a boolean (see `XmlBindingAdapters`).

File: app/src/main/java/org/coepi/android/ui/home/HomeViewModel.kt [hunk 4]
```diff
@@ -21,13 +22,27 @@ import org.coepi.android.ui.symptoms.SymptomsFragmentDirections.Companion.action
...
+    alertsRepo: AlertsRepo
 ) : ViewModel() {
 
     val versionString: LiveData<String> =
         just(resources.getString(home_version, envInfos.appVersionString()))
+            .toLiveData()
+
+    val newAlerts: LiveData<Boolean> = alertsRepo.alerts
+        .map { newAlerts(it.size) }
+        .startWith(newAlerts(0))
```
Comment (i-schuetz) [inline app/src/main/java/org/coepi/android/ui/home/HomeViewModel.kt:35]: This
```
.map { it.isNotEmpty() }
.startWith { false }
```
seems easier to read?

File: app/src/main/java/org/coepi/android/ui/home/HomeViewModel.kt [hunk 5]
```diff
@@ -21,13 +22,27 @@ import org.coepi.android.ui.symptoms.SymptomsFragmentDirections.Companion.action
...
+        .observeOn(AndroidSchedulers.mainThread())
+        .toLiveData()
+
+    val title: LiveData<String> = alertsRepo.alerts
+        .map { title(it.size) }
+        .startWith(title(0))
+        .observeOn(AndroidSchedulers.mainThread())
```
Comment (i-schuetz) [inline app/src/main/java/org/coepi/android/ui/home/HomeViewModel.kt:42]: You can use static import here

File: app/src/main/java/org/coepi/android/ui/home/HomeViewModel.kt [hunk 6]
```diff
@@ -21,13 +22,27 @@ import org.coepi.android.ui.symptoms.SymptomsFragmentDirections.Companion.action
...
-    resources: Resources
+    private val resources: Resources,
+    alertsRepo: AlertsRepo
 ) : ViewModel() {
 
     val versionString: LiveData<String> =
         just(resources.getString(home_version, envInfos.appVersionString()))
+            .toLiveData()
+
+    val newAlerts: LiveData<Boolean> = alertsRepo.alerts
```
Comment (i-schuetz) [inline app/src/main/java/org/coepi/android/ui/home/HomeViewModel.kt:33]: Prepending `has` / `is` is better for booleans.

**Comment by i-schuetz** [comment]
I commented some minor things post-merge. Please submit in a new PR.

**Pull request merged**